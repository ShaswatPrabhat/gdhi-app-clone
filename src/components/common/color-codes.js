export default {
  getColorCodes() {
    // Update variables if changed.
    return [
      { score: "5", color: "#01C975", description: "Most Developed" },
      { score: "4", color: "#80E1CC", description: "" },
      { score: "3", color: "#FFE180", description: "" },
      { score: "2", color: "#FFCA82", description: "" },
      { score: "1", color: "#FCAB9C", description: "Least Developed" },
      { score: "NA", color: "#6C757D" },
    ];
  },
};
