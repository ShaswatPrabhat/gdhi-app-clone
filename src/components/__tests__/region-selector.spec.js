import { shallowMount } from "@vue/test-utils";
import { describe, beforeEach, it, expect, vi } from "vitest";
import RegionSelector from "../regionSelector/region-selector.vue";
import { i18n } from "../../plugins/i18n";
import axios from "axios";
import flushPromises from "flush-promises";
import { EventBus } from "../common/event-bus";
import sinon from "sinon";
describe("Region Selector", () => {
  let wrapper;

  const axiosGetSpy = vi.spyOn(axios, "get");
  const eventBusSpy = vi.spyOn(EventBus, "$emit");
  const responseData = [
    { region_id: "PAHO", regionName: "Pan American Region" },
  ];
  const onChangeStub = sinon.stub();

  const setRegionSpy = vi.fn();

  window.appProperties = {
    setRegion: setRegionSpy,
  };
  const event = { target: { options: { selectedIndex: 1 } } };
  beforeEach(() => {
    axiosGetSpy.mockReset();
    axiosGetSpy.mockResolvedValue({ data: responseData });
    wrapper = shallowMount(RegionSelector, {
      i18n,
    });
  });

  it("should invoke fetch all the regions api", async () => {
    wrapper.vm.fetchRegions();
    await flushPromises();
    expect(axiosGetSpy.mock.calls[0][0]).to.equal("/api/regions");
  });

  it("should render region selector", async () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should emit the event when region dropdown is selected", async () => {
    wrapper.vm.onChange(event);
    expect(eventBusSpy.mock.calls[0][0]).toBe("region:filtered");
  });

  it("should invoke fetch all the regions api for different language", async () => {
    wrapper.vm.selectedRegion = {
      region_id: "PAHO",
      regionName: "Pan American Region",
    };
    wrapper.setMethods({ onChange: onChangeStub });
    wrapper.vm.fetchRegions();
    await flushPromises();
    expect(onChangeStub.called).toBe(true);
  });
});
