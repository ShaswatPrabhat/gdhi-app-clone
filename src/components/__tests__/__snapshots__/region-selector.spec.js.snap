// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Region Selector > should render region selector 1`] = `
<div
  class="region"
  data-v-2c57f00c=""
>
  <div
    class="region-header"
    data-v-2c57f00c=""
  >
    Region
  </div>
  <select
    class="region-select"
    data-v-2c57f00c=""
    name="test_select3"
    style="background-position-x: 95%;"
  >
    <option
      data-v-2c57f00c=""
      value=""
    >
      Select Region
    </option>
    <option
      data-v-2c57f00c=""
      value="PAHO"
    >
       Pan American Region 
    </option>
  </select>
</div>
`;
